graylog2-web-interface/src/components/common/EntityDataTable/EntityDataTable.tsx

```diff
@@ -231,24 +244,31 @@ const EntityDataTable = <Entity extends EntityBase>({
     fixedActionsCellWidth,
   });
 
+  const _setSelectedEntities = useCallback((setSelectedEntitiesArgument: SetStateAction<Array<Entity['id']>>) => {
+    const newState = isFunction(setSelectedEntitiesArgument) ? setSelectedEntitiesArgument(selectedEntities) : setSelectedEntitiesArgument;
+
+    setSelectedEntities(newState);
+    if (onChangeSelection) onChangeSelection(newState);
+  }, [onChangeSelection, selectedEntities]);
+
   const onToggleEntitySelect = useCallback((itemId: string) => {
```

Comment:
Just a small suggestion, we can create a hook like `useSelectedEntities` which can be used like this
```
 const [selectedEntities, setSelectedEntities] = useSelectedEntities(initialSelection, onChangeSelection);
```

It can maintain the state and `_setSelectedEntities`. This way the render function looks a bit cleaner.